### PR TITLE
feat: rename `createProxySSGHelpers` to `createServerSideHelpers`

### LINKED ISSUE #4041: feat: rename `createSSGHelpers`
### Describe the feature you'd like to request

They were never designed only to work with `getStaticProps`. We should have better name for it.

### Describe the solution you'd like to see

- Rename
- Alias to old name with `@deprecated`-notice
- Update docs, especially https://trpc.io/docs/nextjs/ssr that don't mention gSSP at all

### Describe alternate solutions

no

### Additional information

- The new name doesn't have to have `Proxy` in it (that was just to not conflict with interop stuff)
- Don't touch anything in any `**/interop/**`-file

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR implementing this feature!

### PR DESCRIPTION
**Pull request by mozzius**
Closes #4041

## 🎯 Changes

Renamed `createProxySSGHelpers` to `createServerSideHelpers`. Aliased the old name to the new one.
Edited the docs to use the new name.
Changed the examples to use the new name.

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.
- [x] I have come to terms with the fact that I'm bad at using git

### CODE REVIEW

**Comment by mozzius** [comment]
A thought - `createServerSideHelpers` still has to be imported from `@trpc/react-query/ssg`. Should we change this? Might be confusing if you then use it for SSR - perhaps `@trpc/react-query/helpers` would be better

File: www/docs/nextjs/ssr.md [hunk 1]
```diff
@@ -82,6 +82,95 @@ const MyApp: AppType = ({ Component, pageProps }: AppProps) => {
...
+    return {
+      transformer: superjson, // optional - adds superjson serialization
+      links: [
+        httpBatchLink({
+          url: `${getBaseUrl()}/api/trpc`,
+        }),
+      ],
+    },
+  },
+  // `ssr` is false by default
```
Comment (KATT) [inline www/docs/nextjs/ssr.md:103]:
```suggestion
```

**Review by KATT — DISMISSED** [review]

**Review by KATT — APPROVED** [review]

File: www/docs/nextjs/ssr.md [hunk 2]
```diff
@@ -82,6 +82,94 @@ const MyApp: AppType = ({ Component, pageProps }: AppProps) => {
...
+  try {
+    // prefetch `post.byId`
+    await ssg.post.byId.prefetch({ id });
+  } catch (err) {
+    return {
+      // if failed, return 404
+      props: { id },
+      notFound: true,
+    };
+  }
```
[the remark this answers is not in the document]
Comment (mozzius) [reply www/docs/nextjs/ssr.md:143]: This may well be wrong - let me test it myself

[the remark this answers is not in the document]
Comment (mozzius) [reply www/docs/nextjs/ssr.md:143]: Hey - I made a PR to fix this: https://github.com/trpc/trpc/pull/4166

Thanks for informing me, that was a very dumb error on my part.

[the remark this answers is not in the document]
Comment (mozzius) [reply www/docs/nextjs/ssr.md:143]: Here's the preview link to the updated docs:

https://www-git-docs-fix-serversidehelpers-ssr-example-trpc.vercel.app/docs/nextjs/ssr

[the remark this answers is not in the document]
Comment (mozzius) [reply www/docs/nextjs/ssr.md:143]: @shawnmclean the docs have now been fixed. Thanks for pointing this out!

**Pull request merged**